### PR TITLE
More bug fixes

### PR DESCRIPTION
**Pull request by darrennjk**

### CODE REVIEW

File: src/controllers/CinemaController.java [hunk 1]
```diff
@@ -7,90 +7,91 @@
...
-	cinemas.add(newCinema);
-	try {
-		fos = new FileOutputStream(filename);
-		out = new ObjectOutputStream(fos);
-		out.writeObject(cinemas);
-		out.close();
-	} catch (IOException ex) {
-		ex.printStackTrace();
+
+	public void insertCinemaIntoDB(String cinemaName, ArrayList<Movie> movies, HashMap<String, ArrayList<LocalDate>> showtimes) {
```
Comment (darrennjk) [inline src/controllers/CinemaController.java:18]: Renamed the HashMap<**Movie**, ArrayList<LocalDate>> to HashMap<**String**, ArrayList<LocalDate>>

File: src/controllers/CinemaController.java [hunk 2]
```diff
@@ -7,90 +7,91 @@
...
-             ex.printStackTrace();
-         }
-        HashMap<String, ArrayList<Seat>> s = new HashMap<>(); 
-	for (Movie movie : movies) {
-		String title = movie.getTitle();
-		ArrayList<LocalDate> showtime = showtimes.get(title);
-		for (LocalDate sho : showtime) {
-			String key = title + sho.toString();
-			s.put(key, seats);
+	public HashMap<String, ArrayList<Seat>> getSeatsFromDB(ArrayList<Movie> movies, HashMap<String, ArrayList<LocalDate>> showtimes) {
```
Comment (darrennjk) [inline src/controllers/CinemaController.java:42]: Not sure what happened here, doesn't seem like there are any changes

File: src/controllers/CinemaController.java [hunk 3]
```diff
@@ -7,90 +7,91 @@
...
-	try {
-		fis = new FileInputStream(filename);
-		in = new ObjectInputStream(fis);
-		cinemas = (ArrayList<Cinema>) in.readObject();
-		in.close();
-	} catch (IOException ex) {
-		ex.printStackTrace();
-	} catch (ClassNotFoundException ex) {
-		ex.printStackTrace();
+	public void insertSeatIntoDB(SeatType type, Customer customer, String seatNo) {
```
Comment (darrennjk) [inline src/controllers/CinemaController.java:87]: Aligning the arguments to Seat constructor. (Missing seatNo)

File: src/controllers/ClientController.java [hunk 4]
```diff
@@ -38,7 +38,7 @@ public void setFileNameStaff(String fileNameStaff) {
     }
 
     //Inserting Customers name, age, username and password into the database
-    public void insertCustomerToDB(String name, int age, String username, String password, String email, int mobileNumber){
+    public void insertCustomerToDB(String name, int age, String username, String password, String email, String mobileNumber){
```
Comment (darrennjk) [inline src/controllers/ClientController.java]: Changed mobileNumber from type int to String

File: src/controllers/MovieController.java [hunk 5]
```diff
@@ -352,7 +352,7 @@ public boolean addRating(int movieID, String customerName, int numRating, String
 	}
 
 	//Insert a totalSales for a movie into the movie database
-	public boolean addTotalSales(int movieID, double totalSales) {
+	public boolean addTotalSales(int movieID, int totalSales) {
```
Comment (darrennjk) [inline src/controllers/MovieController.java:355]: Changed totalSales from double to int

File: src/controllers/TicketController.java [hunk 6]
```diff
@@ -59,10 +60,10 @@ public void updatePriceToDB(int moviename_index, int cinema_index,int showtime_i
 	}
 	
 	
-	public void insertTicketToDB(Movie a, Cinema b, Customer c, String d, String TID, String custName, String custEmail, String custMobileNumber) {
+	public void insertTicketToDB(Movie movie, Cinema cinema, Customer customer, LocalDate showtime, Seat seat, String TID, String custName, String custEmail, String custMobileNumber) {
```
Comment (darrennjk) [inline src/controllers/TicketController.java:63]: Not sure why the changes were not reflected previously? I thought we had changed to standardize the format.

File: src/entities/Cinema.java [hunk 7]
```diff
@@ -10,12 +10,12 @@ public class Cinema implements Serializable {
     private HashMap<String, ArrayList<LocalDate>> showtimes;
     private HashMap<String, ArrayList<Seat>> seats;  //key is the movie title + showtime.toString().
     private String cinemaName;
-    public Cinema(String cinemaName, ArrayList<Movie> movies, HashMap<Movie, ArrayList<LocalDate>> showtimes, HashMap<String, ArrayList<Seat>> seats) {
+    public Cinema(String cinemaName, ArrayList<Movie> movies, HashMap<String, ArrayList<LocalDate>> showtimes, HashMap<String, ArrayList<Seat>> seats) {
```
Comment (darrennjk) [inline src/entities/Cinema.java:13]: Again, changed Movie to String

File: src/entities/Cinema.java [hunk 8]
```diff
@@ -10,12 +10,12 @@ public class Cinema implements Serializable {
...
         this.movies = movies;
         this.cinemaName = cinemaName;
-        ArrayList<LocalDate> shows = new ArrayList<>();
+        ArrayList<LocalDate> showstime = new ArrayList<>();
         this.showtimes = new HashMap<>();
-        for (Movie mov : showtimes.keySet()) {
+        for (Movie mov : movies) {
```
Comment (darrennjk) [inline src/entities/Cinema.java:18]: This one i'm not too sure what showtimes.keySet() are but changing it to movies fixed the error it had.

File: src/entities/Cinema.java [hunk 9]
```diff
@@ -76,10 +76,9 @@ public Seat getSeat(String title, int index) {
 	return this.seats.get(title).get(index);  
     }
   
-    public void printLayout(Movie movie, LocalDate showtime) { //prints layout of the Cinema based on available seats for the Movie at the particular showtime
+    public void printLayout(String title, LocalDate showtime) { //prints layout of the Cinema based on available seats for the Movie at the particular showtime
```
Comment (darrennjk) [inline src/entities/Cinema.java:79]: Changed from object to String since the method in makeBooking already passes in the title.

File: src/entities/Customer.java [hunk 10]
```diff
@@ -27,7 +27,7 @@ public Customer(String name, int age, String username, String password, String e
     }
 
 
-    public boolean login(ArrayList<Customer> customerDB) throws IOException {
+    public boolean login(ArrayList<Object> customerDB) throws IOException {
```
Comment (darrennjk) [inline src/entities/Customer.java:30]: Since in our interface, we changed the ArrayList to take in Object, we have to update it here as well.

File: src/entities/Customer.java [hunk 11]
```diff
@@ -38,7 +38,7 @@ public boolean login(ArrayList<Customer> customerDB) throws IOException {
         System.out.println("Please enter your password: ");
         tempPassword = reader.readLine();
 
-        ArrayList<Customer> customers = new ArrayList<>();
+        ArrayList<Object> customers = new ArrayList<>();
         customers = customerDB;  //Fetch all customers details from DB
         for (int i = 0; i < customers.size(); i++) {
             if (customers.get(i).getUsername().equals(tempUsername) && customers.get(i).getPassword().equals(tempPassword)) {
```
Comment (darrennjk) [inline src/entities/Customer.java:44]: The issue is over here when we changed to ArrayList<Object>. I think there are some issues with the getters so the getUsername() and getPassword doesn't seem to work.

File: src/entities/Customer.java [hunk 12]
```diff
@@ -177,7 +177,7 @@ private String makeBooking(ArrayList<Cinema> cinemaDB, ArrayList<Movie> movieDB)
 			int col = Integer.parseInt(seatNo.substring(1));
 			int index = row*16+col;
 			cinema.setCustomer(this, title+showtime.toString(), index);
-			Movie mov;
+			Movie mov=null;
```
Comment (darrennjk) [inline src/entities/Customer.java:180]: Have to initialise mov

File: src/entities/Customer.java [hunk 13]
```diff
@@ -188,10 +188,10 @@ private String makeBooking(ArrayList<Cinema> cinemaDB, ArrayList<Movie> movieDB)
 			Ticket newTicket = new Ticket(mov, cinema, this, showtime, cinema.getSeat(title+showtime.toString(), index), TID, this.username, this.email, this.mobileNumber);
 			this.bookings.add(newTicket);
 			System.out.println("The price of your ticket is: " + newTicket.getPrice());
-			System.out.println("Ticket booked successfully. A confirmation has been sent to the provided email and phone number");
 			break;
 		}
-	}		
+	}
+		return "Ticket booked successfully. A confirmation has been sent to the provided email and phone number";
```
Comment (darrennjk) [inline src/entities/Customer.java:194]: This method requires a return method. So I simply converted the last print statement into a return statement.

File: src/entities/Customer.java [hunk 14]
```diff
@@ -207,53 +207,55 @@ private void listMovie(String cinemaname, ArrayList<Cinema> cinemaDB) {
 	}
     }
 
-    private void checkSeatAvailability(ArrayList<Cinema> cinemaDB, ArrayList<Movie> movieDB) {
+    private void checkSeatAvailability(ArrayList<Cinema> cinemaDB, ArrayList<Movie> movieDB) throws IOException {
```
Comment (darrennjk) [inline src/entities/Customer.java:210]: Added throw according to IDE

File: src/entities/Customer.java [hunk 15]
```diff
@@ -207,53 +207,55 @@ private void listMovie(String cinemaname, ArrayList<Cinema> cinemaDB) {
...
-	for (Movie movie : moviesDB) {
-		if (movie.getTitle() == movieTitle) {
-			System.out.println(movie.toString());
-			break;
+		for (Movie movie : moviesDB) {
+			if (movie.getTitle() == movieTitle) {
+				System.out.println(movie.toString());
+				break;
+			}
 		}
```
Comment (darrennjk) [inline src/entities/Customer.java:245]: If im not wrong, there is a missing braces { } around here somewhere but I think it will be easier to debug on the IDE. Can't really seem to pinpoint where it is on GitHub

File: src/entities/Staff.java [hunk 16]
```diff
@@ -15,24 +15,24 @@ public Staff(String username, String password, Cinema cinema) {
...
 	String username = reader.readLine();
 	System.out.println("Enter password: ");
 	String password = reader.readLine();
-	ArrayList<Staff> staffs = staffDB;
+	ArrayList<Object> staffs = staffDB;
 
-	for (Staff s : staffs) {
-		if (s.getUsername(.equals(username) && s.getPassword().equals(password)) {
+	for (Object s : staffs) {
+		if (s.getUsername().equals(username) && s.getPassword().equals(password)) {
```
Comment (darrennjk) [inline src/entities/Staff.java:27]: Missing bracket wasn't reflected as well. And once again, the getUsername and getPassword doesn't work when we change into ArrayList<Object>

File: src/entities/Staff.java [hunk 17]
```diff
@@ -113,7 +113,7 @@ public int compare(Movie o1, Movie o2)
 				System.out.println("Enter Input: ");
 				String searchInput = sc.nextLine();
 				ArrayList<Movie> matchedMovies = null;
-				matchedMovies = mvc.searchMovies(searchInput);
+				matchedMovies = mvc.searchMovies(searchInput); //Still using movie controller. Not sure how to go around this.
```
Comment (darrennjk) [inline src/entities/Staff.java:116]: Not sure how to go about not using mvc here.

File: src/entities/Staff.java [hunk 18]
```diff
@@ -156,7 +156,7 @@ public int compare(Movie o1, Movie o2)
     
     private void createMovieListing() {
 	Scanner sc = new Scanner(System.in);
-	int runtime, option;
+	int runtime, option, id;
```
Comment (darrennjk) [inline src/entities/Staff.java:159]: Added another variable for id, reason below.

File: src/entities/Staff.java [hunk 19]
```diff
@@ -273,8 +276,7 @@ private void createMovieListing() {
 				System.out.println("Invalid date format, Please try again");
 			}
 		}
-
-		Movie newMovie = new Movie(title, type, status, synopsis, director, casts, rating, runtime, releaseDate);
+		Movie newMovie = new Movie(id, title, type, status, synopsis, director, casts, rating, runtime, releaseDate, 0);
```
Comment (darrennjk) [inline src/entities/Staff.java:279]: Had to align to movie constructor since previously had missing arguments. For totalSales, I passed in 0 since I assume for new movies, there should be no sales at all.

File: src/entities/Ticket.java [hunk 20]
```diff
@@ -18,15 +18,14 @@ public class Ticket implements Serializable {
     String holidayFileName;
     Scanner sc= new Scanner(System.in);
 
-    public Ticket(Movie a, Cinema b, Customer c, LocalDate d, Seat seat, String TID, String custName, String custEmail, String custMobileNumber,String holidayFileName) {
-    	movie = a;
-    	cinema = b;
-    	customer = c;
-    	showtime = d;
+    public Ticket(Movie movie, Cinema cinema, Customer customer, LocalDate showtime, Seat seat, String TID, String custName, String custEmail, String custMobileNumber) {
```
Comment (darrennjk) [inline src/entities/Ticket.java:21]: Once again, just standardizing.

File: src/entities/Ticket.java [hunk 21]
```diff
@@ -44,10 +43,10 @@ public void calculatePrice() {
     	double moviePrice;
     	MovieType movieType;
     	int customerAge;
-	SeatType seatType;
+	    SeatType seatType;
     	int date;
     	double multiplier = 1;
-	string day;
+	    String day;
```
Comment (darrennjk) [inline src/entities/Ticket.java]: Capitalized S in string

File: src/entities/Ticket.java [hunk 22]
```diff
@@ -60,9 +59,9 @@ public void calculatePrice() {
     	seatType = seat.getType();
 
         //update multiplier for different days of week
-	//need to create class to calculate holidays
+	    //need to create class to calculate holidays
     	HolidayController a = new HolidayController(holidayFileName);
-	day = showtime.getDayOfWeek().toString()
+	    day = showtime.getDayOfWeek().toString();
```
Comment (darrennjk) [inline src/entities/Ticket.java:65]: Missing ;

File: src/controllers/ClientController.java (hunk 4 above)
Comment (aryans1204) [reply src/controllers/ClientController.java]: uh, we can actually store a 8 digit as an int, because in 32 bits we can store values up to 4 billion (10 digits), and plus changing this would require changes elsewhere too.

File: src/controllers/TicketController.java (hunk 6 above)
Comment (aryans1204) [reply src/controllers/TicketController.java:63]: We hadn't changed before, but its good that you have changed now.

File: src/entities/Cinema.java (hunk 8 above)
Comment (aryans1204) [reply src/entities/Cinema.java:18]: keySet() returns an iterator for the set of keys in the HashMap. It was throwing an error becasue the key in the hashmap is a string of movie title, and not a Movie object, so that was definitely an error.

File: src/entities/Staff.java (hunk 17 above)
Comment (aryans1204) [reply src/entities/Staff.java:116]: We search in the movieDB

**Review by aryans1204 — APPROVED** [review]
Merging these changes for now, though let me know what exactly you mean when you say getUsername() and getPassword() aren't working
